test/BaalSafe.test.ts

```diff
@@ -2528,42 +2526,63 @@ describe("Baal contract", function () {
         summoner.address,
         prop.votingStarts
       );
+      expect(await baal.votingNonces(summoner.address)).to.equal(1);
       expect(priorVotes).to.equal(votes);
       expect(prop.yesVotes).to.equal(votes);
     });
 
+
     it("fail case - fails with different voter", async function () {
+      const expiry = await blockTime() + 1200;
       const signature = await signVote(
         chainId,
         baal.address,
         summoner,
         deploymentConfig.TOKEN_NAME,
+        expiry,
+        0,
         1,
         true
       );
 
       const {v,r,s} = await ethers.utils.splitSignature(signature);
       expect(
-        baal.submitVoteWithSig(applicant.address, 1, true, v, r, s)
+        baal.submitVoteWithSig(applicant.address, expiry, 0, 1, true, v, r, s)
       ).to.be.revertedWith("invalid signature");
+      expect(await baal.votingNonces(applicant.address)).to.equal(0);
     });
 
     it("fail case - cant vote twice", async function () {
+      const expiry = await blockTime() + 1200;
       const signature = await signVote(
         chainId,
         baal.address,
         summoner,
         deploymentConfig.TOKEN_NAME,
+        expiry,
+        0,
         1,
         true
       );
 
       const {v,r,s} = await ethers.utils.splitSignature(signature);
-      await baal.submitVoteWithSig(summoner.address, 1, true, v, r, s);
+      await baal.submitVoteWithSig(summoner.address, expiry, 0, 1, true, v, r, s);
 
+      const signatureTwo = await signVote(
+        chainId,
+        baal.address,
+        summoner,
+        deploymentConfig.TOKEN_NAME,
+        expiry,
+        1,
+        1,
+        true
+      );
+      const sigTwo = await ethers.utils.splitSignature(signatureTwo);
```

Comment:
interesting how this test included a check on nonces inherently as it was using the same signature for both `submitVoteWithSig` - after adding a second signature with the nonce bumped, the test resolved from the error that triggered from adding the nonce. ie notice how nonce had to go from 0 to 1